packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -619,10 +619,13 @@ domain.addBasePathMapping(api1, { basePath: 'go-to-api1' });
 domain.addBasePathMapping(api2, { basePath: 'boom' });
 ```
 
-NOTE: currently, the mapping will always be assigned to the APIs
-`deploymentStage`, which will automatically assigned to the latest API
-deployment. Raise a GitHub issue if you require more granular control over
-mapping base paths to stages.
+You can specify name of API's stage with base path else default mapping will always be assigned to the APIs
+`deploymentStage`, which will automatically assigned to the latest API deployment.
+More info can be found [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-basepathmapping.html#cfn-apigateway-basepathmapping-stage).
+Here is an example to specify stage name:
+```ts
+domain.addBasePathMapping(api1, { basePath: 'go-to-api1', stage: 'prod' });
+```
```

Comment:
```suggestion
You can specify the API `Stage` to which this base path URL will map to. By default, this will be the
`deploymentStage` of the `RestApi`. 

```ts
const betaDeploy = new Deployment(this, 'beta-deployment', {
  api: restapi,
});
const betaStage = new Stage(this, 'beta-stage', {
  deployment: betaDeploy,
});
domain.addBasePathMapping(restapi, { basePath: 'api/beta', stage: betaStage });
```
```